### PR TITLE
handle gnu_source definition for SWIG mapscript (issue #6015)

### PR DESCRIPTION
**Pull request by jmckenna**

### CODE REVIEW

File: mapscript/mapscript.i [hunk 1]
```diff
@@ -33,6 +33,8 @@
 %begin %{
 #ifdef _MSC_VER
 #define SWIG_PYTHON_INTERPRETER_NO_DEBUG
+#else
+#define _GNU_SOURCE 1
```
Comment (sebastic) [inline mapscript/mapscript.i]: Just `#define _GNU_SOURCE` is enough, it doesn't need a value.

Comment (jmckenna) [reply mapscript/mapscript.i]: I did at first but it throws several 'redefined' warnings.  Giving it a value prevents the warnings.

Comment (sebastic) [reply mapscript/mapscript.i]: Then you probably shouldn't define it unconditionally, e.g.:
```C
#ifdef _MSC_VER
#define SWIG_PYTHON_INTERPRETER_NO_DEBUG
#elif !defined(_GNU_SOURCE)
#define _GNU_SOURCE
#endif
```

Comment (jmckenna) [reply mapscript/mapscript.i]: I had tried that earlier as well.  I tried again, still gives warning "/../main/php_config.h:2535:0: warning: _GNU_SOURCE" redefined".

Comment (jmckenna) [reply mapscript/mapscript.i]: With all my testing I found the best way was the pass it a value, to remove the redefinition warnings in PHP source.

Comment (sebastic) [reply mapscript/mapscript.i]: Alright, then just leave it as-is.

Comment (jmckenna) [reply mapscript/mapscript.i]: But I'm open to other ways.  Let me know.

Comment (sebastic) [reply mapscript/mapscript.i]: Looking at [codesearch.debian.net](https://codesearch.debian.net/search?q=%23define+_GNU_SOURCE&literal=0&perpkg=1&page=4) there a quite a number of occurences with value, so either seems acceptable.

Comment (jmckenna) [reply mapscript/mapscript.i]: ok thanks

**Pull request merged**